### PR TITLE
Remove deprecation - Axes, IO, Model

### PR DESCRIPTION
**Pull request by ericpre**
First round of deprecation removal.

### Progress of the PR
- [x] `axes.py`,
- [x] `io.py`,
- [x] `model.py`, 
- [x] update docstring (if appropriate),
- [x] update user guide (if appropriate),
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] update tests,
- [x] ready for review.

### CODE REVIEW

**Review by jlaehne — APPROVED** [review]

File: CHANGES.rst [hunk 1]
```diff
@@ -6,6 +6,60 @@ https://hyperspy.readthedocs.io/en/latest/user_guide/changes.html
 
 .. towncrier-draft-entries:: |release| [UNRELEASED]
 
+
+API Removal
+===========
```
Comment (jlaehne) [inline CHANGES.rst:11]: Shouldn't this be a heading of one level lower than the new release?
(Also applies to the other deprecation PRs)

Comment (jlaehne) [inline CHANGES.rst:11]:
```suggestion
-----------
```

File: CHANGES.rst [hunk 2]
```diff
@@ -6,6 +6,60 @@ https://hyperspy.readthedocs.io/en/latest/user_guide/changes.html
...
+
+Following deprecation cycle, the following methods and functions have been removed:
```
Comment (jlaehne) [inline CHANGES.rst]:
```suggestion
Following the deprecation cycle, the following methods and functions have been removed:
```

File: CHANGES.rst (hunk 1 above)
Comment (ericpre) [reply CHANGES.rst:11]: It doesn't play well with towncrier, so I would suggest to leave it as it is and we edit it during when we do the release.

Comment (jlaehne) [reply CHANGES.rst:11]: Yep, just realized that as well.

File: CHANGES.rst [hunk 3]
```diff
@@ -6,6 +6,63 @@ https://hyperspy.readthedocs.io/en/latest/user_guide/changes.html
 
 .. towncrier-draft-entries:: |release| [UNRELEASED]
 
+
+API Removal
+===========
+
+As the HyperSpy API evolves, APIs are occasionally reorganized or upgraded.
+When APIs evolve, the old API is deprecated and eventually removed in a major
+release. The functions and methods removed in HyperSpy 2.0 is listed below along
```
Comment (jlaehne) [inline CHANGES.rst]:
```suggestion
release. The functions and methods removed in HyperSpy 2.0 are listed below along
```

**Comment by jlaehne** [comment]
I have seen in the RosettaSciIO deprecation PR that not always the changed terminology was properly documented in the user guide. So the documentation should be searched again for all deprecated names in the three deprecation PRs to make sure that there are no remnants of old terminology.

**Comment by ericpre** [comment]
Yes, this should be done after we remove all deprecation?

**Comment by jlaehne** [comment]
> Yes, this should be done after we remove all deprecation?

It could be done in each of the three branches for the respective keywords and functions by searching a local copy (as I did in https://github.com/hyperspy/rosettasciio/pull/30). Otherwise, we should add a checkpoint to the todo list in https://github.com/hyperspy/hyperspy/issues/2996

**Comment by ericpre** [comment]
As there are many deprecations, I made several PRs, so it is not too much pain to review, but the drawback of this approach is that, it is more difficult to find some left over which is why I suggested to do it later.

**Comment by jlaehne** [comment]
OK, I added to the todo list in https://github.com/hyperspy/hyperspy/issues/2996

**Comment by ericpre** [comment]
Sorry, I missed the point that you were referring to the user guide!

**Review by jlaehne — APPROVED** [review]

**Comment by jlaehne** [comment]
I added a PR with some changes to the documentation.

**Review by jlaehne — APPROVED** [review]

**Pull request merged**